src/base/bittorrent/torrentimpl.cpp

```diff
@@ -96,6 +100,102 @@ namespace
+#else
+    TrackerEntry fromNativeAnnouncerEntry(const lt::announce_entry &nativeEntry)
+#endif
+    {
+        TrackerEntry trackerEntry {QString::fromStdString(nativeEntry.url), nativeEntry.tier};
+
+        for (const lt::announce_endpoint &endpoint : nativeEntry.endpoints)
+        {
+#if (LIBTORRENT_VERSION_NUM >= 20000)
+            for (const lt::protocol_version &protocolVersion : {lt::protocol_version::V1, lt::protocol_version::V2})
+            {
+                if (hashes.has(protocolVersion))
+                {
+                    const lt::announce_infohash &infoHash = endpoint.info_hashes[protocolVersion];
+
+                    TrackerEntry::Endpoint trackerEndpoint;
+                    trackerEndpoint.protocolVersion = (protocolVersion == lt::protocol_version::V1) ? 1 : 2;
+                    trackerEndpoint.numSeeds = infoHash.scrape_complete;
+                    trackerEndpoint.numLeeches = infoHash.scrape_incomplete;
+                    trackerEndpoint.numDownloaded = infoHash.scrape_downloaded;
+                    if (infoHash.updating)
+                        trackerEndpoint.status = TrackerEntry::Updating;
+                    else if (infoHash.fails > 0)
+                        trackerEndpoint.status = TrackerEntry::NotWorking;
+                    else if (nativeEntry.verified)
+                        trackerEndpoint.status = TrackerEntry::Working;
+                    trackerEntry.endpoints.append(trackerEndpoint);
+
+                    trackerEntry.numSeeds = std::max(trackerEntry.numSeeds, infoHash.scrape_complete);
+                    trackerEntry.numLeeches = std::max(trackerEntry.numLeeches, infoHash.scrape_incomplete);
+                    trackerEntry.numDownloaded = std::max(trackerEntry.numDownloaded, infoHash.scrape_downloaded);
+                }
+            }
+#else
+            TrackerEntry::Endpoint trackerEndpoint;
+            trackerEndpoint.numSeeds = endpoint.scrape_complete;
+            trackerEndpoint.numLeeches = endpoint.scrape_incomplete;
+            trackerEndpoint.numDownloaded = endpoint.scrape_downloaded;
+            if (endpoint.updating)
+                trackerEndpoint.status = TrackerEntry::Updating;
+            else if (endpoint.fails > 0)
+                trackerEndpoint.status = TrackerEntry::NotWorking;
+            else if (nativeEntry.verified)
+                trackerEndpoint.status = TrackerEntry::Working;
+            trackerEntry.endpoints.append(trackerEndpoint);
+
+            trackerEntry.numSeeds = std::max(trackerEntry.numSeeds, endpoint.scrape_complete);
+            trackerEntry.numLeeches = std::max(trackerEntry.numLeeches, endpoint.scrape_incomplete);
+            trackerEntry.numDownloaded = std::max(trackerEntry.numDownloaded, endpoint.scrape_downloaded);
+#endif
+        }
+
+        const auto [numUpdating, numWorking, numNotWorking, numNotContacted] =
```

Comment:
You should track the variables in the loop above.
Also you only need 3 `bool` variables.